### DISCUSSION TITLE
CLI access an opened database

### DISCUSSION THREAD

**Discussion by krodelabestiole** [keepassxreboot/keepassxc#9823] [Q&A] [answered]
Hi, I use keepassXC on GNOME for all my passwords. I also use it to add keys to the SSH agent, and for freedsktop secret service.
Opening my database is the first thing I do after starting my computer.

I guess I could also use keepass for sudo, or to enter some secrets in some scripts. I think keepass-cli could be useful in this regard.
But it is not as long as I have to enter my main password every time I need to access any secret data !

Is there any way to extract secret data from a database that is currently opened in keepassXC, without having to enter any password ?
or am I missing something ?

**Comment by droidmonkey** [comment, marked as answer]
No, they are completely separate programs. There are requests in to support this, but it is not a priority for us at this time.

**Reply by laur89** [reply]
Can't find open issue for this. Could you please point to one so interested parties can follow it?

**Reply by droidmonkey** [reply]
https://github.com/keepassxreboot/keepassxc/issues/5069

**Comment by waynew** [comment]
It seems like it would be relatively trivial to add connections to [the proxy](https://github.com/varjolintu/keepassxc-proxy/blob/master/NativeMessagingProxy.cpp) to just work like the browser extension does. I don't (yet) know what all the syntax is, but that wouldn't require any changes on the main side.
